core/src/main/scala/precog/AutoBump.scala

```diff
@@ -140,12 +154,16 @@ object AutoBump {
   }
 
   /** Extract change label from trickleUpdateDependencies log */
-  def extractLabel(lines: List[String]): ChangeLabel = {
+  def extractLabel(lines: List[String]): IO[Either[Warnings, ChangeLabel]] = IO {
```

Comment:
BTW, I think the only reason this is in `IO` is because `ChangeLabel.unapply` can throw an error. From a design standpoint, you would usually want to represent that error state with an `Either` if possible. Not a big deal here, just a note.

Reply:
Actually, it's a partial function so no error will be thrown. There's actually no reason for it being `IO` -- I just started out by assuming it would need to be, and never questioned it.